python/cudf/pyproject.toml

```diff
@@ -123,6 +124,12 @@ skip = [
     "__init__.py",
 ]
 
+[tool.pytest.ini_options]
+filterwarnings = [
+    # some third-party dependencies (e.g. 'boto3') still using datetime.datetime.utcnow()
+    "ignore:.*datetime.*utcnow.*scheduled for removal:DeprecationWarning",
+]
```

Comment:
In the `cudf` and `dask_cudf` tests on Python 3.12, there are a few dozen cases where `DeprecationWarning` like the following are treated as errors

> DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).

([build link](https://github.com/rapidsai/cudf/actions/runs/10724599183/job/29742782194#step:7:4834))

Looking through the logs, those all appear to be coming from code in `boto3` / `botocore` that we don't control. There are no direct uses of `datetime.datetime.utcnow()` in this repo.

```shell
git grep utcnow
```

So this is proposing not failing tests when those warnings are raised.

Context on the deprecation of `datetime.datetime.utcnow()`:

* https://blog.miguelgrinberg.com/post/it-s-time-for-a-change-datetime-utcnow-is-now-deprecated
* https://docs.python.org/3/library/datetime.html#datetime.datetime.utcnow